index.src.html

```diff
@@ -769,103 +782,123 @@ <h3 id=interface-reporting-observer>Interface {{ReportingObserver}}</h3>
+
+  6. <a>Queue a task</a> to [[#invoke-observers]] with a list containing the <a>context object</a>
+
   The <dfn method for=ReportingObserver><code>disconnect()</code></dfn> method,
   when invoked, must run these steps:
-  
-  1. If present, remove the <a>context object</a> from the list of
-     <a>registered reporting observers</a> for the <a>environment settings
-     object</a> associated with the <a>context object</a>.
- 
+
+  1. If the <a>context object</a> is not <a>registered</a>, stop.
+
+  2. Remove the <a>context object</a> from the <a>registered reporting observer
+     list</a> of its associated <a>environment settings object</a>.
+
+  <h3 id="add-report" algorithm>
+    Add |report| to reporting |observer|
+  </h3>
+
+  This algorithm adds |report| as a {{Report}} object to |observer|'s <a>report
+  list</a>, so long as |report|'s <a>type</a> is observable by |observer|.
+
+  1. If |observer|'s <a>options</a> has a {{ReportingObserverOptions/types}}
+     member which does not contain |report|'s [=report/type=], stop.
+
+  2. Create a new {{Report}} |r| with {{Report/type}} initialized to |report|'s
+     [=report/type=], {{Report/url}} initialized to |report|'s [=report/url=],
+     and {{Report/body}} initialized to |report|'s [=report/body=].
+
+  Issue: how to polymorphically initialize body?
+
+  3. Add |r| to the end of |observer|'s <a>report list</a>.
+      
   <h3 id="notify-observers" algorithm>
     Notify reporting observers on |settings| with |report|
   </h3>
   
-  This algorithm makes |report|'s contents available to any registered
-  {{ReportingObserver}}s on the provided <a>environment settings object</a>
+  This algorithm makes |report|'s contents available to any <a>registered</a>
+  <a>reporting observers</a> on the provided <a>environment settings object</a>
   |settings|.
   
-  1. For each {{ReportingObserver}} |observer| of the <a>registered
-     observers</a> associated with |settings|:
-     
-     1. If |observer|'s <a>options</a> has a {{ReportingObserverOptions/types}}
-        member which does not contain |report|'s [=report/type=], skip these
-        sub-steps.
-        
-     2. Create a new {{Report}} |r| with {{Report/type}} initialized to
-        |report|'s [=report/type=], {{Report/url}} initialized to |report|'s
-        [=report/url=], and {{Report/body}} initialized to |report|'s
-        [=report/body=].
-
-        Issue: how to polymorphically initialize body?
-       
-     3. Add |r| to the end of |observer|'s <a>report list</a>.
-         
-  2. <a>Queue a task</a> to [[#invoke-observers]] with a copy of the
-     <a>registered observers</a> associated with |settings|.
+  1. For each {{ReportingObserver}} |observer| <a>registered</a> with
+     |settings|, execute [[#add-report]] on |report| and |observer|.
+ 
+  2. Add |report| to the <a>report buffer</a> associated with |settings|.
+  
+  3. If the <a>report buffer</a> now contains more than 100 reports, remove the
```

Comment:
"The oldest report" implies that we're keeping timestamps or some other accounting information -- can you say something like 'remove the item at the beginning of report buffer'? (Since I think they're guaranteed to be stored in chronological order)

Reply:
Okay, yeah.